### PR TITLE
net: shell: add conn_mgr shell commands

### PR DESCRIPTION
**Pull request by glarsennordic**
Add shell commands to controll conn_mgr

### CODE REVIEW

**Comment by glarsennordic** [comment]
Automated tests and docs for this shell addition will be a separate task, but it does pass manual testing

**Review by krish2718 — DISMISSED** [review]
based on a quick review, LGTM.

File: subsys/net/lib/shell/cm.c
Comment (rlubos) [inline subsys/net/lib/shell/cm.c:47]: `const`?

File: subsys/net/lib/shell/cm.c
Comment (rlubos) [inline subsys/net/lib/shell/cm.c:174]: nit: `target->iface == NULL` (misra). No need for an empty line before check.

File: subsys/net/lib/shell/cm.c
Comment (rlubos) [inline subsys/net/lib/shell/cm.c:145]: ditto

File: subsys/net/lib/shell/cm.c
Comment (jukkar) [inline subsys/net/lib/shell/cm.c:57]: Do we need this fancy short name handling? The shell has support for dynamic shell completion so user does not need to type full string.

Comment (glarsennordic) [reply subsys/net/lib/shell/cm.c:57]: It is probably not needed. I will look into what would be required for dynamic shell completion. But I would guess that would be similarly complex to implement, since Zephyr needs to be made aware of the possible auto-completions

Comment (glarsennordic) [reply subsys/net/lib/shell/cm.c:57]: To take advantage of dynamic command completion would require a significant or complete restructuring of this command set, which I am not eager to undertake.

For the time being, I will simply remove the flag name aliases

File: subsys/net/lib/shell/cm.c
Comment (jukkar) [inline subsys/net/lib/shell/cm.c:178]: alignment

File: subsys/net/lib/shell/cm.c
Comment (jukkar) [inline subsys/net/lib/shell/cm.c:256]: alignment

**Review by jukkar — COMMENTED** [review]
About the dynamic completion, adding for example network interface dynamic completion should be somewhat easy, both the `net iface` (in iface.c) and `net stats` (in stats.c) commands have enabled that and they share the code for doing it. Not sure how well it would suit `cm` command. Anyway, this can be enhanced later too if needed.

**Comment by glarsennordic** [comment]
@jukkar This is, I think, ready for merge (once it has your approval), but Robert is OOO, do you know who I should contact to move forward with merge?

**Review by krish2718 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**
